assignment2/q1.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+struct Node{
+    int value;
+    Node *parent;
+    Node *left, *right;
+};
+
+Node *FindKey(Node *root, int key) {
+
+    if(root->value == key)
+        return root;
+
+    Node *l = NULL;
+    Node *r = NULL;
+
+    if(root->left != NULL) {
+        l = FindKey(root->left, key);
+        if(l != NULL)
+            return l;
+    }
+    if(root->right != NULL) {
+        r = FindKey(root->right, key);
+        if(r != NULL)
+            return r;
+    }
+
+    return NULL;
+}
+
+void PrintAncestors(Node *root, int key) {
+
+    if(root == NULL) {
+        cout << "The tree is empty." << endl;
+        return;
+    }
+    Node* descendant = FindKey(root, key);
+
+    if(descendant != NULL) {
+        cout << "Descendant key is "<< descendant->value <<endl;
+        if(descendant->parent == NULL)
+            cout<< "NULL";
```

Comment:
In case there are no ancestors, it is ok, in my opinion, to print an 'empty list', i.e. print nothing.
Can skip this print in my opinion (the `while` already handles the no-ancestors case).